### ISSUE TITLE
DoubleHistogram doesn't set autoResize on underlying AbstractHistogram

### ISSUE THREAD

**Issue by colings86** [HdrHistogram/HdrHistogram#67]
If an empty `DoubleHistogram` is created with `autoResize` set to `true` and then this is used to merge other `DoubleHistogram`s into it, it throws an `ArrayOutOfBoundsException` from the underlying `AbstractHistogram`. Given that the `DoubleHistogram` is set to auto-resize I would think the underlying `AbstractHistogram` should auto-resize as well?

Code to reproduce:

```java
package sandbox;

import org.HdrHistogram.DoubleHistogram;

public class HDRHistoTest {

    public static void main(String[] args) {

        DoubleHistogram histo1 = new DoubleHistogram(3);
        histo1.setAutoResize(true);
        histo1.recordValue(6.0);
        histo1.recordValue(1.0);
        histo1.recordValue(5.0);
        histo1.recordValue(8.0);
        histo1.recordValue(3.0);
        histo1.recordValue(7.0);
        DoubleHistogram histo2 = new DoubleHistogram(3);
        histo2.setAutoResize(true);
        histo2.recordValue(9.0);
        DoubleHistogram histo3 = new DoubleHistogram(3);
        histo3.setAutoResize(true);
        histo3.recordValue(4.0);
        histo3.recordValue(2.0);
        histo3.recordValue(10.0);

        DoubleHistogram merged = new DoubleHistogram(3);
        merged.setAutoResize(true);
        merged.add(histo1);
        merged.add(histo2);
        merged.add(histo3);
        System.out.println("SUCCESS!");
    }

}

```

This code throws the `ArrayOutOfBoundsException` at `merged.add(histo1);`

**Comment by giltene** [comment]
Addressed with changes introduced in 2.1.5. Added testAddWithAutoResize  for coverage with https://github.com/HdrHistogram/HdrHistogram/commit/cc107c69a5cc4893a2abc762808d006acbb63086 (fails on 2.1.4, works with 2.1.5).

**Comment by colings86** [comment]
Thanks for the quick fix and release :)
